services:
  db:
    image: mariadb:11
    container_name: ordenavia-db
    restart: always
    volumes:
      - mariadb_data:/var/lib/mysql
    environment:
      MARIADB_ROOT_PASSWORD: ${DB_ROOT_PASSWORD}
      MARIADB_DATABASE: ${DB_NAME}
      MARIADB_USER: ${DB_USER}
      MARIADB_PASSWORD: ${DB_PASSWORD}
    ports:
      - "${DB_PORT}:3306"
    healthcheck:
      test: ["CMD-SHELL", "mariadb -hlocalhost -u${DB_USER} -p${DB_PASSWORD} -e 'SELECT 1;' ${DB_NAME}"]
      interval: 5s
      timeout: 10s
      retries: 5
    networks:
      ordenavia-net:
        ipv4_address: 172.25.0.2

  backend:
    build:
      context: ./backend
      dockerfile: Dockerfile.prod
      args:
        - PYTHON_VERSION=${PYTHON_VERSION}
    env_file:
      - ./backend/.env
    depends_on:
      db:
        condition: service_healthy
    expose:
      - 8000
    command: gunicorn ordenavia.wsgi:application --bind 0.0.0.0:8000 --workers 3
    volumes:
      - backend_static_volume:/app/staticfiles
      - backend_media_volume:/app/media
      - /srv/ordenavia/backend_staticfiles:/app/staticfiles
      - /srv/ordenavia/backend_media:/app/media
    networks:
      ordenavia-net:
        ipv4_address: 172.25.0.10

  frontend:
    build:
      context: ./frontend
      dockerfile: Dockerfile.prod
    env_file:
      - ./frontend/.env
    expose:
      - 80
    networks:
      ordenavia-net:
        ipv4_address: 172.25.0.11

volumes:
  mariadb_data:
  backend_static_volume:
  backend_media_volume:

networks:
  ordenavia-net:
    driver: bridge
    ipam:
      config:
        - subnet: 172.25.0.0/16
          gateway: 172.25.0.1
